apollo-router/src/spec/query.rs

```diff
@@ -465,13 +465,15 @@ impl Query {
                 _ => Ok(()),
             },
             executable::Type::Named(name) if name == "Int" => {
-                let opt = if input.is_i64() {
-                    input.as_i64().and_then(|i| i32::try_from(i).ok())
-                } else if input.is_u64() {
-                    input.as_i64().and_then(|i| i32::try_from(i).ok())
-                } else {
-                    None
-                };
+                let opt = input.as_i64();
+
+                if opt.is_some_and(|i| i32::try_from(i).is_err()) {
+                    tracing::warn!(
+                        path = %Path::from_response_slice(path),
```

Comment:
The `Display` implementation for this type does write out the index of arrays. Just something to be aware of if we're concerned about the cardinality of any metric we create out of that path property.

Reply:
@TannerRogalsky updated... haven't actually tested the code :D let me what you think.